xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultModelBridge.java

```diff
@@ -179,20 +179,33 @@ public boolean copy(DocumentReference source, DocumentReference destination)
 
     @Override
     public boolean delete(DocumentReference reference)
+    {
+        return delete(reference, true);
+    }
+
+    @Override
+    public boolean expurge(DocumentReference reference)
+    {
+        return delete(reference, false);
+    }
+
+    private boolean delete(DocumentReference reference, boolean totrash)
     {
         XWikiContext xcontext = this.xcontextProvider.get();
         try {
             XWikiDocument document = xcontext.getWiki().getDocument(reference, xcontext);
             if (document.getTranslation() == 1) {
-                xcontext.getWiki().deleteDocument(document, xcontext);
-                this.logger.info("Document [{}] has been deleted.", reference);
+                xcontext.getWiki().deleteDocument(document, totrash, xcontext);
+                this.logger.info("Document [{}] has been deleted (send to trash: [{}]).", reference, totrash);
```

Comment:
Same comment

Reply:
I'm using `toRecycleBin` for the new code. I propose to migrate the `toTrash` to `toRecyclebin` gradually.